### DISCUSSION TITLE
Question about logic level voltage

### DISCUSSION THREAD

**Discussion by dburr** [erichelgeson/BlueSCSI-v1#84] [Q&A]
Hi, just out of curiosity, My understanding is that SCSI signaling is at 5v. But I thought the STM32 was a 3.3v device. Yet I don't see any sort of logic level conversion going on anywhere on the BlueSCSI. Other SCSI emulators such as the RASCSI (which uses a Raspberry Pi, also a 3.3V device) have the signals run through logic level conversion chips (SN74LS641-1's, in the case of the RASCSI.) Is there something I'm missing here?

**Comment by erichelgeson** [comment]
The pins used on the stm32 are 5v tolerant, see https://github.com/erichelgeson/BlueSCSI/blob/main/ArdSCSinoV2.png

Some info on SCSI signaling:

![scsisignal](https://user-images.githubusercontent.com/271734/161401776-68ce2361-93fd-45d0-be66-6edda433617d.png)
https://www.ti.com/lit/an/slla035/slla035.pdf

Full disclosure I'm just a hobbyist learning, so please let me know if I made a mistake - that's how we all learn!

Note that TERMPOWR is 5v - and that is used to drive the pullup/pulldown passive termination.
